### PR TITLE
Fix parsing and tests

### PR DESCRIPTION
**Pull request by uklotzde**

### CODE REVIEW

File: src/library/searchquery.cpp [hunk 1]
```diff
@@ -12,8 +12,16 @@
 #include "util/db/sqllikewildcards.h"
 
 namespace {
-const QRegularExpression kDurationRegex(QStringLiteral("^(\\d*)(m|:)?([0-6]?\\d)?s?$"));
-const QRegularExpression kNumericOperatorRegex(QStringLiteral("^(>|>=|=|<|<=)(.*)$"));
+const QRegularExpression kDurationRegex(QStringLiteral("^(\\d+)(m|:)?([0-5]?\\d)?s?$"));
```
Comment (Be-ing) [inline src/library/searchquery.cpp:15]: Was the change from `0-6` to `0-5` intentional?

**Comment by Be-ing** [comment]
Thank you for digging into this.

File: src/library/searchquery.cpp (hunk 1 above)
Comment (uklotzde) [reply src/library/searchquery.cpp:15]: Sure, :00-:59 is the valid range. I also added a test that checks that :60 is not accepted.

Comment (uklotzde) [reply src/library/searchquery.cpp:15]: The regex was wrong and no one ever noticed.

**Pull request merged**